docker-compose.yml

```diff
@@ -1,13 +1,18 @@
-version: '3'
+# necessary for volumes_from, single-host oriented
+version: '2'
```

Comment:
Hmm, same as https://github.com/libero/browser/blob/11d1d58563fa95f56528b0fa8969177e68e7875c/docker-compose.yaml#L26-L32?

Reply:
I'll try, but custom paths are not documented for `volumes_from`

Reply:
One workaround would be to change the folder to `/app/browser/public/` in `browser`'s container image, to keep it unique.

Reply:
Doesn't seem great to pin us to an older version. Maybe time to avoid mounting and extend the image?

Reply:
[This often-linked maintainer's comment](https://github.com/docker/compose/issues/4693#issuecomment-291652194) somewhat officially states that v2 is the best version to use if `docker stack` the (multi-node) version isn't being used. This however is not reported in the [official documentation](https://docs.docker.com/compose/compose-file/compose-versioning/) and v2 seems not to have been updated since 2018 started.

Kubernetes supports [sharing volumes between containers](https://kubernetes.io/docs/tasks/access-application-cluster/communicate-containers-same-pod-shared-volume/) since the concept of Pod implies co-located containers. That would probably need a separate `initContainer` for initializing that volume, but still makes sharing volumes a first-class concepts.

Other open source PHP applications that publishes Docker images may be biased by their history. Wordpress for example has both `apache` and `fpm` images, I wonder how the `fpm` ones work?

Reply:
(Wordpress may use a named volume: https://github.com/mjstealey/wordpress-nginx-docker/blob/master/docker-compose.yml)

Reply:
MediaWiki only provides a [Apache-based image](https://github.com/wikimedia/mediawiki-docker/blob/master/1.31/Dockerfile).
[Various](https://github.com/runeasgar/docker_drupal_stack/blob/master/docker-compose.yml#L12) [Drupal](https://github.com/traktion/drupal-docker-compose/blob/master/docker-compose.yml#L11) [examples](https://github.com/industrydive/docker-drupal/blob/master/docker-compose.yml#L14) use `volumes_from`.
Joomla examples use both [`volumes_from`](https://github.com/tritanium-industries/TITDev-Docker/blob/185beab28e6da6ddcd84fbeb2c0f843990561979/docker-compose.yml#L15) or [named volumes](https://github.com/xbx/docker-joomla-behind-proxy-example/blob/9f01b4f49dc70a1856dc5f9b0123d5e88a8e5d9c/docker-compose.yml#L29).

So the options are:

- docker-compose v2 format: `volumes_from`, updates automatically, uncertain long-term shelf life
- docker-compose v2 format: named volumes, updates have to destroy (some) volumes explicitly
- Kubernetes via Minikube to deploy on a single node: lots of overhead, uncertain solution

Reply:
To keep it the same as elsewhere, we could add a `docker-compose -f docker-compose.yaml down --v` at https://github.com/libero/environments/blob/eaaa46fb23262e8408605403face63de996404a2/scripts/remote-deploy.sh#L30? This isn't ideal, extending images or Kubernetes is preferable but can be introduced in the future.

Reply:
It is a strong assumption that all services are stateless, which they are for now

Reply:
https://github.com/libero/environments/pull/9 add this